pwndbg/dbg/__init__.py

```diff
@@ -5,14 +5,47 @@
 from __future__ import annotations
 
 from typing import Any
+from typing import Callable
+from typing import List
 from typing import Tuple
 
 dbg: Debugger = None
 
 
+class Session:
+    """
+    Interactive debugger session. Handles things like commands and history.
+    """
+
+    def history(self) -> List[str]:
+        """
+        The command history of this interactive session.
```

Comment:
This comment makes it seem like you get the whole history, but GDB will only give you the last 10 commands. It also sounds like each element is the history command, but the format is `<index> <command>`, not even mentioning whitespace. How does LLDB do this? We should either clarify in this comment what the user is going to get back (regardless of debugger), or update the implementation in `GDBSession` to match the expected format.

Reply:
That's a fair point. I had originally intended to just do as little as needed to move this out, while also not breaking any existing code, and leave sorting this out to a future PR. But we might as well do it now.

LLDB can show the full history, in a format that is close, but not compatible for the purposes of `pwndbg.commands`. So I feel it would make sense to both change this method so that it returns the last N (defaults to 10) entries in the history, as a list of tuples <index, command>, and change the code in `GDBSession` and the code in `check_repeated` to reflect that.